### PR TITLE
Improve settings test coverage for tauri-app

### PR DESCRIPTION
**Pull request by hardingjam**
## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs
- [ ] included screenshots (if this involves a front-end change)

## Summary by CodeRabbit

- **Tests**
  - Replaced the previous settings test suite with a new, comprehensive test file.
  - Enhanced coverage for reactive store behaviors related to accounts, networks, orderbooks, and subgraphs.
  - Validated synchronization, filtering, resetting, and edge cases in settings-related state management.
  - Improved test isolation and robustness for settings features.

### CODE REVIEW

**Comment by hardingjam** [comment]
> @hardingjam I think we should move the tests from settings into it's own file. This is an issue I've fixed today with this PR #1749
> 
> I was getting blank page in tauri app becase vitest dependency is used in the prod build and that was causing problems. Moving the tests to a dedicated file fixed the issue so I think we can do the same here

Updated the PR with your separate test file

**Review by hardyjosh — APPROVED** [review]

File: tauri-app/src/lib/test/settings.test.ts [hunk 1]
```diff
@@ -1,111 +1,444 @@
...
+			expect(get(activeNetworkRef)).toBe('mainnet');
+		});
+
+		test('resetActiveNetworkRef should set undefined when no networks', async () => {
+			const emptySettings = { ...mockConfigSource, networks: {} };
+			settings.set(emptySettings);
+
+			await resetActiveNetworkRef();
+
+			expect(get(activeNetworkRef)).toBeUndefined();
```
[the remark this answers is not in the document]
Comment (hardingjam) [reply tauri-app/src/lib/test/settings.test.ts]: done
